cardano-crypto-praos/src/Cardano/Crypto/VRF/Praos.hs

```diff
@@ -311,42 +308,35 @@ mkSignKey = fmap SignKey $ newForeignPtr finalizerFree =<< mallocBytes signKeySi
 mkProof :: IO Proof
 mkProof = fmap Proof $ newForeignPtr finalizerFree =<< mallocBytes (certSizeVRF)
 
-proofFromBytes :: ByteString -> Proof
-proofFromBytes bs
-  | BS.length bs /= certSizeVRF
-  = error "Invalid proof length"
-  | otherwise
-  = unsafePerformIO $ do
+proofFromBytes :: ByteString -> Either String Proof
+proofFromBytes bs = case BS.length bs of
+  bsLen -> if bsLen /= certSizeVRF
+    then Left $ "Invalid proof length " <> show @Int bsLen <> ", expecting " <> show @Int certSizeVRF
+    else Right . unsafePerformIO $ do
```

Comment:
Since we will not call these directly from any `Decoder` the only caller will be from the "raw deserialise" functions then we do not need to use Either, since the only caller uses Maybe. So we don't need the message construction here.

Reply:
Is there a case for changing the caller to use `Either` instead of `Maybe`?